src/wp-admin/includes/image.php

```diff
@@ -1336,28 +1359,34 @@ function _copy_image_file( $attachment_id ) {
  * For example an `image/jpeg` should be converted into an `image/jpeg` and `image/webp`. The first type
  * is considered the primary output type for this image.
  *
+ * Called for each uploaded image to determine the list of mime types that should be converted into. Then,
+ * called again for each target image size as they are generated to see if the image should be converted into the mime type
+ * for that size.
+ *
  * @since 6.1.0
  *
- * @param $attachment_id int The attachment ID.
+ * @param int    $attachment_id  The attachment ID.
+ * @param string $image_size     The image size name. False when called on the image and target size is unavailable.
  * @return array An array of valid mime types, where the key is the source file mime type and the list of mime types to
  *               generate.
  */
-function wp_upload_image_mime_transforms( $attachment_id ) {
+function wp_upload_image_mime_transforms( $attachment_id, $image_size ) {
 	$image_mime_transforms = array(
 		'image/jpeg' => array( 'image/jpeg', 'image/webp' ),
 		'image/webp' => array( 'image/webp', 'image/jpeg' ),
 	);
 
 	/**
-	 * Filter to the output mime types for a given input mime type.
+	 * Filter to the output mime types for a given input mime type and image size.
 	 *
 	 * @since 6.1.0
 	 *
-	 * @param array $image_mime_transforms A map with the valid mime transforms where the key is the source file mime type
-	 *                                     and the value is one or more mime file types to generate.
-	 * @param int   $attachment_id         The ID of the attachment where the hook was dispatched.
+	 * @param array  $image_mime_transforms A map with the valid mime transforms where the key is the source file mime type
+	 *                                      and the value is one or more mime file types to generate.
+	 * @param int    $attachment_id         The ID of the attachment where the hook was dispatched.
+	 * @param string $image_size            The image size name. Optional.
 	 */
-	return (array) apply_filters( 'wp_upload_image_mime_transforms', $image_mime_transforms, $attachment_id );
+	return (array) apply_filters( 'wp_upload_image_mime_transforms', $image_mime_transforms, $attachment_id, $image_size );
```

Comment:
As in the Performance plugin, we removed the casting value returned by webp_uploads_upload_image_mime_transforms to an array in PR [#393](https://github.com/WordPress/performance/pull/393). If it makes sense, can we remove it from the WordPress implementation?

Reply:
Good point, thanks for the feedback @mukeshpanchal27 - I'll update that here as well

Reply:
updated